Plugins/Editor/Scripts/Control/Helpers/CracksStitching.cs

```diff
@@ -0,0 +1,541 @@
+        /// <returns> Yield while solving if the preprocessor has been enabled, otherwise returns empty </returns>
+        public static IEnumerable SolveRaw(Vector3[] vertices, int[] tris, List<int>[] subMeshes, ISolverDebugProvider debug, float maxDist = 0.05f, CancellationToken pCancellationToken = default)
+        {
+            // Merging duplicate vertices to ignore material-specific topology
+            Merge(vertices, out var newVertices, ref tris);
+            
+            pCancellationToken.ThrowIfCancellationRequested();
+            
+            var nativeVertices = vertices;
+            vertices = newVertices;
+
+            var allEdges = new HashSet<EdgeId>();
+            var sharedEdges = new HashSet<EdgeId>();
+            
+            // We're thinking of cracks as edges that do not have multiple triangles 
+            for (int i = 0; i < tris.Length; i+=3)
+            {
+                int x = tris[i], y = tris[i + 1], z = tris[i + 2];
+                
+                var edgeA = new EdgeId(x, y);
+                var edgeB = new EdgeId(y, z);
+                var edgeC = new EdgeId(z, x);
+                
+                if (allEdges.Add(edgeA) == false)
+                    sharedEdges.Add(edgeA);
+                if(allEdges.Add(edgeB) == false)
+                    sharedEdges.Add(edgeB);
+                if(allEdges.Add(edgeC) == false)
+                    sharedEdges.Add(edgeC);
+            }
+            
+            pCancellationToken.ThrowIfCancellationRequested();
+            
+            // Only keep edges which do not share multiple triangles
+            var leftToSolve = new HashSet<EdgeId>(allEdges);
+            foreach (var edge in sharedEdges)
+                leftToSolve.Remove(edge);
+
+            var trianglesToAdd = new List<(int a, int b, int c)>();
+            var workingData = new WorkingData(trianglesToAdd, allEdges, leftToSolve, vertices);
+            
+            debug?.HookIntoWorkingData(workingData);
+
+            while (leftToSolve.Count > 0)
+            {
+                pCancellationToken.ThrowIfCancellationRequested();
+                
+                // Take one random edge from the hashset
+                EdgeId thisEdge;
+                using (var e = leftToSolve.GetEnumerator())
+                {
+                    e.MoveNext();
+                    thisEdge = e.Current;
+                }
+
+                if (ReturnBestMatchFor(ref thisEdge, ref workingData, out var bestMatch) == false || bestMatch.dist > maxDist)
+                {
+                    if(bestMatch.dist <= maxDist)
+                        throw new InvalidOperationException($"Stray edge ({thisEdge}) could not be solved for");
+                    
+                    debug?.LogWarning($"For edge {thisEdge}, closest match ({bestMatch.edge}) did not satisfy {nameof(maxDist)} constraint {bestMatch.dist}/{maxDist}");
```

Comment:
Should be using `String.Combine` instead of string interpolation, as older versions of Unity do not support it.